src/main/java/org/dflib/jjava/ExtensionLoadingMaster.java

```diff
@@ -0,0 +1,44 @@
+package org.dflib.jjava;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class ExtensionLoadingMaster {
+
+    private static ExtensionLoadingMaster instance;
+
+    private final Set<String> usedExtensions;
+
+    private ExtensionLoadingMaster() {
+        usedExtensions = new HashSet<>();
+    }
+
+    public Iterable<Extension> getExtensions(String jarPath) {
+        try (URLClassLoader classLoader = URLClassLoader.newInstance(new URL[]{Path.of(jarPath).toUri().toURL()})) {
```

Comment:
Can't we do this like in one go, by collecting all the jars first and then just grab all the extensions without filtering them?